src/components/documentManager.ts

```diff
@@ -0,0 +1,778 @@
+     */
+    private static preloadImage(src: string | null): Promise<string | null> {
+        if (src) {
+            return new Promise((resolve, reject) => {
+                const preload = new Image();
+                preload.src = src;
+                preload.addEventListener('load', () => {
+                    resolve(src);
+                });
+                preload.addEventListener('error', () => {
+                    // might also resolve and return null here, to have the caller take away the background.
+                    reject();
+                });
+            });
+        } else {
+            return Promise.resolve(null);
+        }
+    }
+
+    /**
+     * Get url for primary image for a given item
+     *
+     * @param {BaseItemDto} item to look up
+     * @returns {Promise<string | null>} url to image after preload
+     */
+    private static getPrimaryImageUrl(
+        item: BaseItemDto
+    ): Promise<string | null> {
+        let src: string | null = null;
+
+        if (item.AlbumPrimaryImageTag && item.AlbumId) {
+            src = JellyfinApi.createImageUrl(
+                item.AlbumId,
+                'Primary',
+                item.AlbumPrimaryImageTag
+            );
+        } else if (item.ImageTags?.Primary && item.Id) {
+            src = JellyfinApi.createImageUrl(
+                item.Id,
+                'Primary',
+                item.ImageTags.Primary
+            );
+        }
+
+        if (
+            item?.UserData?.PlayedPercentage &&
+            item?.UserData?.PlayedPercentage < 100 &&
+            !item.IsFolder &&
+            src != null
+        ) {
+            src += `&PercentPlayed=${item.UserData.PlayedPercentage}`;
+        }
+
+        return this.preloadImage(src);
+    }
+
+    /**
+     * Get url for logo image for a given item
+     *
+     * @param {BaseItemDto} item to look up
+     * @returns {Promise<string | null>} url to logo image after preload
+     */
+    private static getLogoUrl(item: BaseItemDto): Promise<string | null> {
+        let src: string | null = null;
+        if (item.ImageTags?.Logo && item.Id) {
+            src = JellyfinApi.createImageUrl(
+                item.Id,
+                'Logo',
+                item.ImageTags.Logo
+            );
+        } else if (item.ParentLogoItemId && item.ParentLogoImageTag) {
+            src = JellyfinApi.createImageUrl(
+                item.ParentLogoItemId,
+                'Logo',
+                item.ParentLogoImageTag
+            );
+        }
+
+        return this.preloadImage(src);
+    }
+
+    /**
+     * This fucntion takes an item and shows details about it
+     * on the details page. This happens when no media is playing,
+     * and the connected client is browsing the library.
+     *
+     * @param {BaseItemDto} item to show information about
+     * @returns {Promise<void>} for the page to load
```

Comment:
JSDoc syntax is like this: @param {type} argumentInFunction - Description

@param {type} [argumentInFunction=value] - Description

for arguments that have default values.

Non-blocking though, I can fix that in the linting PR.

Reply:
As far as I can see the hyphen is optional and just a readability thing, but from this I can fix the other one